src/textfile_gui.cpp

```diff
@@ -344,48 +339,48 @@ static void Xunzip(byte **bufp, size_t *sizep)
 	FILE *handle = FioFOpenFile(textfile, "rb", dir, &filesize);
 	if (handle == nullptr) return;
 
-	this->text = ReallocT(this->text, filesize);
-	size_t read = fread(this->text, 1, filesize, handle);
+	char *buf = MallocT<char>(filesize);
+	size_t read = fread(buf, 1, filesize, handle);
 	fclose(handle);
 
 	if (read != filesize) return;
 
 #if defined(WITH_ZLIB)
 	/* In-place gunzip */
-	if (StrEndsWith(textfile, ".gz")) Gunzip((byte**)&this->text, &filesize);
+	if (StrEndsWith(textfile, ".gz")) Gunzip((byte**)&buf, &filesize);
 #endif
 
 #if defined(WITH_LIBLZMA)
 	/* In-place xunzip */
-	if (StrEndsWith(textfile, ".xz")) Xunzip((byte**)&this->text, &filesize);
+	if (StrEndsWith(textfile, ".xz")) Xunzip((byte**)&buf, &filesize);
 #endif
 
-	if (!this->text) return;
+	if (buf == nullptr) return;
 
-	/* Add space for trailing \0 */
-	this->text = ReallocT(this->text, filesize + 1);
-	this->text[filesize] = '\0';
+	this->text = std::string(buf, filesize);
+	free(buf);
 
 	/* Replace tabs and line feeds with a space since StrMakeValidInPlace removes those. */
-	for (char *p = this->text; *p != '\0'; p++) {
-		if (*p == '\t' || *p == '\r') *p = ' ';
+	for (auto &c : this->text) {
+		if (c == '\t' || c == '\r') c = ' ';
 	}
```

Comment:
Could be done with `std::transform`, although that might not be as clear/terse as this code.

Reply:
You answered yourself! :D